src/main/resources/config/liquibase/changelog/20241206114231_changelog.xml

```diff
@@ -0,0 +1,51 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<databaseChangeLog xmlns="http://www.liquibase.org/xml/ns/dbchangelog"
+                   xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+                   xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog https://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-latest.xsd">
+    <changeSet id="20241206114231" author="rangger">
+        <preConditions onFail="MARK_RAN">
+            <not>
+                <tableExists tableName="file_upload"/>
+            </not>
+        </preConditions>
+
+        <createTable tableName="file_upload">
+            <column name="id" type="bigint" autoIncrement="true">
+                <constraints primaryKey="true" nullable="false"/>
+            </column>
+
+            <column name="path" type="varchar(255)">
+                <constraints nullable="false"/>
+            </column>
+
+            <column name="server_file_path" type="varchar(255)">
+                <constraints nullable="false"/>
+            </column>
+
+            <column name="filename" type="varchar(512)">
+                <constraints nullable="false"/>
+            </column>
+
+            <column name="entity_id" type="bigint">
+                <constraints nullable="true"/>
+            </column>
```

Comment:
does it really make sense to allow this to be nullable? When do file uploads happen, which are not connected to any entity? Imo that doesn't really make sense, and that would also obsolete the checks and deletion scheduled tasks in the other files

Reply:
You are right for some cases, but I do think in various cases this might still happen. Consider you want to create a single user-facing form to create an entity that has files and e.g. a name. In case you want to create the entity first, you must submit the form with the name and the file uploads together to create the entity server side and then the file upload entries. This might lead to issues like requests exceeding the request size limit. If we keep it nullable, we can have the form to create the entity, create a single request for every single file and simply attach the id's of the uploaded files to the form-request to create the entity.

An occurence where this might happen for example is the creation of a lecture, where you can drag and drop files into the description of the lecture, which get uploaded and stored before the lecture entity itself exists.

Hope this makes sense to you. 😅